ddsketch/ddsketch.go

```diff
@@ -203,16 +204,36 @@ func (s *DDSketch) ToProto() *sketchpb.DDSketch {
 	}
 }
 
-// FromProto builds a new instance of DDSketch based on the provided protobuf representation.
+// FromProto builds a new instance of DDSketch based on the provided protobuf representation, using a Dense store.
 func FromProto(pb *sketchpb.DDSketch) (*DDSketch, error) {
+	return FromProtoWithStoreType(pb, store.Dense)
+}
+
+// FromProtoWithStoreType builds a new instance of DDSketch based on the provided protobuf representation, with the provided store type
+func FromProtoWithStoreType(pb *sketchpb.DDSketch, storeType store.Type) (*DDSketch, error) {
```

Comment:
I have mixed feelings about `store.Type`. I don't think this pattern helps with modularity (e.g., it's not convenient to use with custom store implementations, and as of now, collapsing stores do not have their own types), it cannot be used with parameterized stores, it does not allow reusing stores (e.g., store pools), etc.

Instead, what do you think of something along those lines:

```
type StoreProvider func() Store

var (
	BufferedPaginatedStoreConstructor = StoreProvider(func() Store { return NewBufferedPaginatedStore() })
	...
)
```

Reply:
good point, agreed. I've implemented it with the `Provider` you suggested.